heron/spi/src/java/com/twitter/heron/spi/statemgr/SchedulerStateManagerAdaptor.java

```diff
@@ -209,6 +209,68 @@ public Boolean deleteSchedulerLocation(String topologyName) {
   }
 
   /**
+   * Checks to see if the execution state exists
+   *
+   * @return a boolean indicating whether the execution state exists or not
+   */
+  public boolean doesExecutionStateExist(String topologyName) {
+    return awaitResult(delegate.executionStateExists(topologyName));
+  }
+
+  /**
+   * Clean all states of a heron topology.
+   * TMasterLocation, PackingPlan, PhysicalPlan, SchedulerLocation, ExecutionState, and Topology
+   */
+  public Boolean cleanState(String topologyName) {
+    LOG.fine("Cleaning up topology state");
+
+    Boolean result;
+
+    // It is possible that  TMasterLocation, PhysicalPlan and SchedulerLocation are not set
+    // Just log but don't consider them failure
+    result = deleteTMasterLocation(topologyName);
```

Comment:
It is deleted a few lines below this (https://github.com/twitter/heron/pull/1128/files#diff-6564229e92c7d6359355807e020d0972R237)